.github/workflows/release.yml

```diff
@@ -1,16 +1,165 @@
+# **Artifacts**: java libraries and bill of materials
+# **Artifact repository**: https://maven.forgerock.org/artifactory/community/com/forgerock/sapi/gateway/
 on:
-  release:
-    types: [published]
+  workflow_dispatch:
+    inputs:
+      notes:
+        description: "Release notes"
+        required: false
+        type: string
+        default: ''
+      is_final_release:
+        description: "Is a final release, otherwise will be a 'rc' (release candidate)"
+        required: false
+        type: boolean
+        default: false
+      candidate_number:
+        description: "Provide the candidate sequence number (default: 1)"
+        required: false
+        type: string
+        default: 1
+env:
+  GH_TOKEN: ${{ github.token }}
+  GIT_AUTHOR_NAME: ${{ secrets.GIT_COMMIT_USERNAME_BOT }}
+  GIT_AUTHOR_EMAIL: ${{ secrets.GIT_COMMIT_AUTHOR_EMAIL_BOT }}
+  GIT_COMMITTER_NAME: ${{ secrets.GIT_COMMIT_USERNAME_BOT }}
+  GIT_COMMITTER_EMAIL: ${{ secrets.GIT_COMMIT_AUTHOR_EMAIL_BOT }}
+  GIT_SHA_SHORT: $(echo ${{ github.sha }} | cut -c1-7)
+
 jobs:
-  publish:
+
+  release_prepare: # prepare for a release in scm, creates the tag and release branch with the proper release versions
+    name: Call release prepare
+    # uses: ./.github/workflows/release-prepare.yml # same repository
+    uses: SecureApiGateway/secure-api-gateway-parent/.github/workflows/release-prepare.yml@issue/812-action-release-prepare-workflow
+    with:
+      is_final_release: ${{ inputs.is_final_release }}
+      candidate_number: ${{ inputs.candidate_number }}
+    secrets: # inherit only when is the same repository
+      GPG_PRIVATE_KEY_BOT: ${{ secrets.GPG_PRIVATE_KEY_BOT }}
+      GPG_KEY_PASSPHRASE_BOT: ${{ secrets.GPG_KEY_PASSPHRASE_BOT }}
+      GIT_COMMIT_USERNAME_BOT: ${{ secrets.GIT_COMMIT_USERNAME_BOT }}
+      GIT_COMMIT_AUTHOR_EMAIL_BOT: ${{ secrets.GIT_COMMIT_AUTHOR_EMAIL_BOT }}
+
+  release_perform:
     runs-on: ubuntu-latest
-    name: Deploy release
+    name: Build and publish the release
+    needs: release_prepare
     steps:
+      - name: outputs context
+        run: |
+          echo "release ref: ${{ needs.release_prepare.outputs.release_ref }}"
+          echo "release_branch: ${{ needs.release_prepare.outputs.release_branch }}"
+          echo "tag_ref: ${{ needs.release_prepare.outputs.tag_ref }}"
+
+      # https://github.com/actions/checkout
       - uses: actions/checkout@v3
+        id: checkout
+        name: checkout master
+        with:
+          ref: ${{ needs.release_prepare.outputs.release_branch }}
+
+      # https://github.com/crazy-max/ghaction-import-gpg
+      # https://httgp.com/signing-commits-in-github-actions/
+      - name: Import GPG key
+        id: gpg
+        uses: crazy-max/ghaction-import-gpg@v5
         with:
-          ref: ${{ github.ref }}
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY_BOT }}
+          passphrase: ${{ secrets.GPG_KEY_PASSPHRASE_BOT }}
```

Comment:
Who's GPG account is this?

Why can't we just do a gpg --keyserver  --recv-keys <kid> instead of using this dodgy sounding gh action?